Diff/Diff.sublime-syntax

```diff
@@ -1,75 +1,408 @@
 %YAML 1.2
 ---
-# http://www.sublimetext.com/docs/3/syntax.html
+# https://www.sublimetext.com/docs/syntax.html
+# https://www.gnu.org/software/diffutils/manual/diffutils.html
 name: Diff
+scope: source.diff
+version: 2
+
 file_extensions:
   - diff
   - patch
+
 first_line_match: |-
-  (?x)^
-      (===\ modified\ file
-      |==== \s* // .+ \s - \s .+ \s+ ====
-      |Index:[ ]
-      |---\ [^%]
-      |\*\*\*.*\d{4}\s*$
-      |\d+(,\d+)* (a|d|c) \d+(,\d+)* $
-      |diff\ --git[ ]
-      )
+  ^(?x:
+    ===\ modified\ file
+  | ==== \s* // .+ \s - \s .+ \s+ ====
+  | Index:[ ]
+  | Prereq:[ ]
+  | ---\ [^%]
+  | \*\*\*.*\d{4}\s*$
+  | \d+(,\d+)* [adc] \d+(,\d+)* $
+  )
 
-scope: source.diff
 contexts:
   main:
-    - match: '^((\*{15})|(={67})|(-{3}))$\n?'
-      scope: meta.separator.diff
+    - match: ^(?={{git_first_line}})
+      push: Packages/Git Formats/Git Diff.sublime-syntax#email-first-line
+    - include: leading-text
+    - include: line-ranges
+    - include: headers
+    - include: deltas
+    - include: diff3-normal
+
+###[ LEADING TEXT ]############################################################
+
+  leading-text:
+    - match: ^(Index)(:)[ \t]*
       captures:
-        1: punctuation.definition.separator.diff
-    - match: ^\d+(,\d+)*(a|d|c)\d+(,\d+)*$\n?
-      scope: meta.diff.range.normal meta.range.normal.diff
-    - match: ^(@@)\s*(.+?)\s*(@@)\s*(.*?)\s*$\n?
-      scope: meta.diff.range.unified meta.range.unified.diff
-      captures:
-        1: punctuation.definition.range.diff
-        2: meta.toc-list.line-number.diff
-        3: punctuation.definition.range.diff
-        4: entity.name.section.diff
-    - match: '^(((\-{3}) .+ (\-{4}))|((\*{3}) .+ (\*{4})))$\n?'
-      scope: meta.diff.range.context meta.range.context.diff
-      captures:
-        3: punctuation.definition.range.diff
-        4: punctuation.definition.range.diff
-        6: punctuation.definition.range.diff
-        7: punctuation.definition.range.diff
-    - match: '(^(((-{3}) .+)|((\*{3}) .+))$\n?|^(={4}) .+(?= - ))'
+        1: meta.mapping.key.diff keyword.other.diff
+        2: punctuation.separator.key-value.diff
+      push: index-body
+    - match: ^(Prereq)(:)[ \t]*
+      captures:
+        1: meta.mapping.key.diff keyword.other.diff
+        2: punctuation.separator.key-value.diff
+      push: prereq-body
+
+  index-body: 
+    - meta_scope: meta.mapping.diff meta.diff.index meta.index.diff
```

Comment:
https://github.com/sublimehq/Packages/blob/b2d49371e0845034b0e89c556811ff1475afdab2/Diff/Diff.sublime-syntax#L72

Backward compatibility. 

Various color schemes target `meta.diff.[header|index|range]`. Removing them would cause all those color schemes to break.